go/enclave/evm/evm_facade.go

```diff
@@ -94,18 +93,22 @@ func ExecuteOffChainCall(msg *types.Message, s *state.StateDB, header *common.He
 	vmenv := vm.NewEVM(blockContext, txContext, s, chainConfig, vmCfg)
 
 	result, err := gethcore.ApplyMessage(vmenv, msg, gp)
-	if err != nil {
-		// also return the result as the result can be evaluated on some errors like ErrIntrinsicGas
-		logger.Error("ErrKey applying msg:", log.ErrKey, err)
-		return result, fmt.Errorf("unable to ApplyMessage - %w", err)
-	}
-
+	// Follow the same error check structure as in geth
+	// 1 - vmError / stateDB err check
+	// 2 - evm.Cancelled() TODO
+	// 3 - error check the ApplyMessage
 	// Read the error stored in the database.
 	err = s.Error()
 	if err != nil {
 		return nil, newErrorWithReasonAndCode(err)
 	}
 
+	if err != nil {
```

Comment:
this block should be after the revert, and the "err" should prob be called something else

Reply:
still, this check needs to go last I'd say. to be consistent with geth

Reply:
Changed the name of the dbErr 👍 

I thought it was high-to-low error checking. VM, then Execution, then Result err checking. But I might have misunderstood that.

Reply:
I don't think geth checks for Result reverse in this method, it's only when it's returning the payload.
https://github.com/ethereum/go-ethereum/blob/5a02b2d6d0a8d5c8ebf170505c681f7b19df7ed9/internal/ethapi/api.go#L1049

But logically speaking is the same thing, so I think it's fine to have this order of checks.